src/matrixClass.ts

```diff
@@ -0,0 +1,317 @@
+                        if(this.ships[1])this.ships[1] = false;
+                        else this.ships[2] = false;
+                    }
+                    else if(hitCount == 4) this.ships[3] = false;
+                    else if(hitCount == 5) this.ships[4] = false;
+
+
+
+                    hitCount = 0 ;
+                }
+            }
+        }
+    }
+
+        public surroundVerticalShips(){
+        let hitCount;
+        for(let col =0; col < this.board.length; col ++){
+            hitCount = 0;
+            for(let row = 0; row < this.board.length; row ++){
+                if(this.board[row][col] == 1)hitCount ++;
+                else if(this.board[row][col]!= 1){
+                    if(hitCount > 1)this.verticalSurroundShip(row,col,hitCount);
+                    
+                    if(hitCount == 2){
+                        this.ships[0] = false;
+                    }
+                    else if(hitCount == 3){
+                        if(this.ships[1])this.ships[1] = false;
+                        else this.ships[2] = false;
+                    }
+                    else if(hitCount == 4) this.ships[3] = false;
+                    else if(hitCount == 5) this.ships[4] = false;
+
+                    hitCount = 0 ;
+                }
+            }
+        }
+    }
+
+    private horizontalSurroundShip(row:number, col:number, length)
+    {
+        for(let i =0; i <= length+1; i ++ ){
+            if(col-i >= 0 && col-i<=9){
+                if(row-1>= 0)this.board[row-1][col-i] =4;
+                this.board[row][col-i] =4;
+                if(row+1<10)this.board[row+1][col-i] =4;
+            }
+        }
+    }
+
+    private verticalSurroundShip(row:number, col:number, length)
+    {
+        for(let i =0; i <= length+1; i ++ ){
+            if(row-i>=0 && row-i<=9){
+                if(col-1>=0)this.board[row-i][col-1] =4;
+                this.board[row-i][col] =4;
+                if(col+1<10)this.board[row-i][col+1] =4;
+            }
+        }
+    }
+
+
+    public isThereUnsunkShipAt(lastShot:Position):boolean
+    {
+        // as a side affect add a 3 where the shot should be if trued
+        lastShot.print("unsunk " );
+        var up = lastShot.getPositionAbove();
+        var down = lastShot.getPositionDown();
+        var left = lastShot.getPositionLeft();
+        var right = lastShot.getPositionRight();
+        if(left.Column<9 && this.alreadyHitAt(left) || right.Column>0 && this.alreadyHitAt(right)){
+            //horizontal ship
+            console.log("HORIZONTAL ")
+            return this.horizontalShipSunk(lastShot);
+        }
+        else if (this.getRowPos(up.Row) <9  && this.alreadyHitAt(up) || (this.getRowPos(down.Row)>0 && this.alreadyHitAt(down))){
+            //vert ship
+            console.log("vertical ")
+            return this.verticalShipSunk(lastShot);
+        }
+        else{
+            console.log("ship size 1")
+            //ship of size one
```

Comment:
Again, if you've got these comments, you probably wanted to change something, code-wise, since you couldn't understand it without them!

Why not have a `return this.handleHorizontalShip(params)` call instead of the bit inside.

Then you can have a `handleVerticalShip` and a `handleSizeOneShip` method as well, and cut down the length of this class loads.